### PR TITLE
[TorchInductor] Canonicalize torch.concatenate in normalization_pass

### LINKED ISSUE #194898: [TorchInductor] normalization_pass misses torch.concatenate
### 🐛 Describe the bug

`normalization_pass` canonicalizes  `torch.concat`, but misses the `torch.concatenate` form.

## Minimal reproducer

```python
from __future__ import annotations

import torch


class Model(torch.nn.Module):
    def forward(self, x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
        return torch.concatenate([x, y], dim=-1)


def make_case(device: torch.device):
    torch.manual_seed(1010)
    model = Model().to(device).eval()
    inputs = (
        torch.randn(2, 16, device=device, requires_grad=True),
        torch.randn(2, 16, device=device, requires_grad=True),
    )
    return model, inputs


if __name__ == "__main__":
    device = torch.device("cuda:0")

    model, inputs = make_case(device)

    with torch.no_grad():
        eager = model(*inputs)

        torch._dynamo.reset()

        with torch._inductor.config.patch(
            pre_grad_fusion_options={"normalization_pass": {}}
        ):
            compiled = torch.compile(
                model,
                fullgraph=True,
            )
            actual = compiled(*inputs)
```

The same graph written with `torch.concat` is canonicalized

```python
class Model(torch.nn.Module):
    def forward(self, x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
        return torch.concat([x, y], dim=-1)
```

After `normalization_pass`, it becomes

```python
cat = torch.cat([x, y], dim=1)
```

## Observed behavior

The traces will generate FX graph.

```bash
TORCH_TRACE=trace_path python test.py
tlparse trace_path -o tlparse_path
```

For `torch.concatenate`, both `before_pre_grad_graph` and `after_pre_grad_graph` still contain:

```python
concatenate = torch.concatenate([x, y], dim=-1)
```

## Downstream impact

This is not only a canonicalization difference. It can also block a subsequent pre-grad optimization.

For example, with both `normalization_pass` and `split_cat_pass` enabled, the following `torch.concat` form:

```python
torch.concat(list(torch.split(x, 2, dim=-1)), dim=-1)
```

is first canonicalized to `torch.cat`, allowing `split_cat_pass` to recognize the redundant split-and-reassemble pattern and eliminate it. If the same program is written with `torch.concatenate` instead:

```python
torch.concatenate(list(torch.split(x, 2, dim=-1)), dim=-1)
```

`normalization_pass` leaves the `torch.concatenate` node unchanged, and the subsequent `split_cat_pass` does not match it. 

## Root cause

`torch.concatenate` is not included in the `CallFunctionVarArgs` pattern. 

Reproduced on PyTorch main commit `e289091`.

### Versions

PyTorch: 2.15.0a0+gite289091
Python: 3.12.13
Ubuntu 22.04.5
CUDA 12.9
NVIDIA RTX 4090

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

### PR DESCRIPTION
**Pull request by Ultron09**
# Fixing an Issue

Before submitting, please review:
- [The Ultimate Guide to PyTorch Contributions](https://github.com/pytorch/pytorch/wiki/The-Ultimate-Guide-to-PyTorch-Contributions#getting-started-with-pull-requests)
- [AI-Assisted Development](https://github.com/pytorch/pytorch/blob/main/AI_POLICY.md) policy

---

## Issue

Fixes #194898

## Summary

In `normalization_pass` (`torch/_inductor/fx_passes/split_cat.py`), `CallFunctionVarArgs` was registered for `[torch.cat, torch.concat]`, but missed the `torch.concatenate` alias. Consequently, models using `torch.concatenate` left un-canonicalized `torch.concatenate` nodes in the FX graph, which prevented subsequent pre-grad optimizations (such as `split_cat_pass`, `merge_splits_pass`, etc.) from matching and optimizing them.

This PR adds `torch.concatenate` to `CallFunctionVarArgs` in `normalize_cat_default` and adds test coverage for `torch.concatenate` and negative dimension canonicalization in `test_split_cat_fx_passes.py`.

## Checklist

- [x] Passes lint (`spin fixlint`)
- [x] Added/updated tests
- [ ] Updated documentation (if applicable)
- [ ] Included benchmark results (for PRs impacting perf)

## BC-breaking?

No

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Review by jansel — APPROVED** [review]
